module/vision/GoalDetector/src/GoalDetector.cpp

```diff
@@ -293,12 +293,25 @@ namespace module::vision {
                                 goals->goals.push_back(std::move(g));
                             }
                         }
+
+                        // Grab the point closest in the cluster and assume it is the bottom of the post
+                        auto closest_uPCw_idx =
+                            std::min_element(cluster.begin(), cluster.end(), [&](const int& a, const int& b) {
+                                return rays.col(a).z() < rays.col(b).z();
+                            });
+                        Eigen::Vector3d closest_uPCw = Eigen::Vector3d(rays.col(*closest_uPCw_idx).cast<double>());
+                        // Project goal point onto the field plane
+                        auto Hwc = horizon.Hcw.inverse();
+                        Eigen::Vector3d rGWw =
+                            closest_uPCw * std::abs(Hwc.translation().z() / closest_uPCw.z()) + Hwc.translation();
+                        goals->rGWw.push_back(rGWw);
+                        log<NUClear::DEBUG>("Goal at: ", rGWw.transpose());
```

Comment:
How come this is all separate to the old method? Shouldn't this method be replacing the goal measurement in the individual Goal message?